### ISSUE TITLE
Group for "other" tags

### ISSUE THREAD

**Issue by lonix1** [Fooxly/vscode-provision-bar#8]
Hey @thom1606 sorry for bombarding you with all these issues, just installed this extension and trying it out. Loving it so far!! :+1:

A minor discussion point about `provision.groups`:

Let's say I have these tags: BUG, TODO, NOTE, FOO, BAR. I created keywords for the most important tags: BUG, TODO, NOTE. I didn't create keywords for FOO and BAR (and BAZ, QUX, etc., etc.) because the statusbar would become cluttered.

I read the docs for `provision.groups` and assumed it was for "other" tags, so I added FOO and BAR as a group. But that didn't show. The group only shows tags which are **already defined** as keywords (maybe docs should say that?).

Obviously I read the docs incorrectly, but this got me thinking that this behaviour would be nice. Instead of cluttering the statusbar with a million tags, rather only show the important ones, and stuff the rest into an "OTHER" group - the way I assumed above.

What do you think? Feel free to close if you don't like this idea. Thanks again for this nice extension!

**Comment by thom1606** [comment]
Hey there,
you are currently not understanding the usage of keywords and groups I think.
Every keyword which you want to recognise in our ProVision packages has to be noted inside the provision.keywords property.

You can then go ahead and create a group for keywords to group them together. For example, TODO and FIXME could join in an "Warnings" group. Each keyword which does not have an group is then simply not shown.

We are currently rewriting the core of ProVision to make this easier and will consider rewriting the documentation for this. Thanks for pointing this out. Hopefully this helps you in the meantime!

**Comment by lonix1** [comment]
> Each keyword which does not have an group is then simply not shown.

Thanks, that was the bit I didn't understand! Although I would say that *any keyword which is part of a group is only shown in that group; other keywords are shown independently.*

So for anyone who stumbles on this, here is an example that works:
````jsonc
"provision.keywords": {
  // will be shown independently:
  "FOO":  { /* ... */ },
  "BAR":  { /* ... */ },
  "BAZ":  { /* ... */ },
  // will be shown as part of "OTHER" group:
  "SPAM": { /* ... */ },
  "HAM":  { /* ... */ },
  "EGGS": { /* ... */ },
},
"provision.groups": [{
  "keywords": [ "SPAM", "HAM", "EGGS" ],
  "title": { "*": "OTHER" },
  /* ... */
}],
````

Thanks for this extension, it's really useful! :+1:

**Comment by lonix1** [comment]
After the update to 4, it stopped working. The config changed from "provision" to "ProVision". But after I fixed that, it still didn't work.

How do I convert the old config (above) to the new config?

**Comment by thom1606** [comment]
Hey,

Could you maybe take a look at our manual via the help: ProVision command? There we have explained everything. If you are unable to get it to work, please try to reach out again.

Sorry for the inconvenience!

**Comment by lonix1** [comment]
I did that already, I removed all my old settings from `.vscode/settings.json` and replaced them with what I copied from the docs:
````json
  "ProVision.groups": {
    "notes": {
      "title": "$(info) {0}",
      "tooltip": "All my notes",
      "foregroundStyle": "statusBarItem.warningForeground",
      "backgroundStyle": "statusBarItem.warningBackground"
    },
  },
  "ProVision.keywords": {
    "TODO": {
      "group": "notes",
      "color": "#FFFFFF",
      "backgroundColor": "#F2B01F",
      "overviewRulerColor": "rgba(242, 176, 31, 0.8)",
      "overviewRulerLane": 4
    },
    "FIXME": {
      "group": "notes",
      "color": "#FFFFFF",
      "backgroundColor": "#D85F88",
      "overviewRulerColor": "rgba(216, 95, 136, 0.8)",
      "overviewRulerLane": 4
    },
    "NOTE": {
      "group": "notes",
      "color": "#AAAAAA",
      "backgroundColor": "#434343",
      "overviewRulerColor": "rgba(67, 67, 67, 0.8)",
      "overviewRulerLane": 4
    },
  },
````

But I couldn't even get that to work. What did I do wrong?

**Comment by thom1606** [comment]
Hey,
I have a couple of questions for you:
- does it work when you remove all your settings and let ProVision use it default settings?
- Do you have a note inside your current active file?

**Comment by lonix1** [comment]
- no
- yes

I will tinker with it some more and let you know what I discover.

**Comment by thom1606** [comment]
could you maybe share a screenshot of the file you are testing?

**Comment by lonix1** [comment]
Not sure how to do that, but it has all the various combinations
- //TODO foo
- // TODO foo
- //NOTE foo
- // NOTE foo
- etc.

If it works for you, then problem is probably on my side. I'll carry on investigating.

Are there any logs I can look into?

**Comment by thom1606** [comment]
Hey there,
good news, I have found the problem for you, by default keywords have a prop called "mustIncludeColon" and "caseSensitive" which are true by default. If you set these to false it should work for you.

We found out this is not in our manual and will update this within a couple of days! Sorry for the inconvenience.

**Comment by lonix1** [comment]
There are many breaking changes - there should be noted in the release notes so others are warned before they upgrade.

Also various things don't work anymore. e.g. 
"ProVision.bar.side": "left", / "right" For me it's always on the extreme right
and
"ProVision.bar.priority": xxx, This setting doesn't change anything
and
if there is no group, then nothing shows (unlike before), there must be groups
and
it doesn't show per file like before, now its for the whole workspace
and
if a keyword doesn't belong to a group, it won't be shown independently like before

UPDATE:
Some things can be forced to work by restarting vscode, I can't just edit the config and save.

Maybe I should open a separate issue?

**Comment by thom1606** [comment]
Hey,
we have tested the issues you send and found the following:

**Side and priority issues**
We have tested the side and priority and it works for us, maybe your priority number isn't high enough? For the side we can't really explain why that is happing to you, the default value is left, and we have tried changing the setting but it always get's to the right side. (we only port left and right to the vscode api.)

**Keywords not shown without a group**
Items with no group are not shown, this is correct. Because ProVision also supports syntax highlighting, and people do not want every keyword to be tracked by the ProVision bar. We have chosen to only show them if a group is added to the keyword. (groups don't need to be defined in the ProVision.groups property because that is just for styling). The only other option was to add a boolean value but this keeps the fields to a minimum.

**It doesn't show per file**
We don't really understand this. ProVision itself works upon a file based manner and only searches inside the currently active file. So we don't really know how this can be shown per workspace? Also, we are not able to reproduce this.

**If a keyword doesn't belong to a group, it doesn't show up**
Like I said before, we have chosen to make the group property mandatory if you want them to show up in the Status Bar. So to achieve this, you simply need to create a custom group inside the keyword to get this to work.

We will be updating the package within the following week to address some issues and test this all some more. But we hope this explains some things.
